### ISSUE TITLE
Empty bytes in buffer after partial read

### ISSUE THREAD

**Issue by Michael-F-Ellis** [kgrz/reading-files-in-go#2]
Thanks for creating this. It was helpful but I think there's a problem if more than one chunk of data is read. I found it necessary to add a counter for nread and trim the data after the for loop.

```	
	data = make([]byte, 4096)
	datalen := 0
	for {
		nread, err = f.Read(data)
		datalen += nread
		if err != nil {
			if err != io.EOF {
				return data, fmt.Errorf("failed reading data from file : %v", err)
			}
			break // reached EOF
		}
	}
	data = data[0:datalen] 
```

**Comment by kgrz** [comment]
Indeed! Thanks for pointing this out. I'm assuming you're referring to https://github.com/kgrz/reading-files-in-go/blob/0764d352fe5fd83584c66c0a91611e89ddb6f4d7/reading-chunkwise.go#L21

I'm not getting the total, fixed in bf1b2f3

**Comment by kgrz** [comment]
Reverted that commit. I think I misunderstood. Can you tell me what's the file size in the example you've seen? In the file I've pointed out the reading is done at 100 byte chunks at a time, and the output is printed inside the loop itself, so there may not be a need for another accumulator. 

It does, however, not work if the loop simply reads and the buffer size is grown at every loop iteration, which is not the example/post shows. Am I missing anything else?
